controllers/charmedk8scontrolplane_controller.go

```diff
@@ -430,14 +497,22 @@ func (r *CharmedK8sControlPlaneReconciler) reconcileDelete(ctx context.Context,
 		}
 	}
 
-	// TODO: clean up secrets for kubeconfig once that is implemented
+	kubeConfigSecret := &kcore.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cluster.Namespace,
+			Name:      cluster.Name + "-kubeconfig",
+		},
+	}
+	if err := r.Client.Delete(ctx, kubeConfigSecret); err != nil && !apierrors.IsNotFound(err) {
+		log.Error(err, "failed to delete kubeconfig secret", "secret", kubeConfigSecret.Name)
+	}
```

Comment:
Should err be returned here? As it stands, we'll move on to... a regular requeue, by the looks of it.

Reply:
I guess I'm worried about corner cases in reconcileDelete in general. If there are no machines, but there is a kubeconfig secret, then the function will remove the finalizer and exit without deleting the kubeconfig secret.

Reply:
adjusted the sequence of deletion. Now kubeconfig is the first thing that gets checked in the reconcileDelete call, this way it will be deleted regardless of machine existence. Also return the error in addition to logging it